modules/Module.py

```diff
@@ -1,22 +1,26 @@
 from lucifer.Utils import RetrieveShell
 
 
-class BaseModule(RetrieveShell):
-    def __init__(self, luciferManager, ShellRun=False):
+class BaseModule(RetrieveShell):   # All Modules should inherit from BaseModule (this class)
+    def __init__(self, luciferManager, ShellRun=False):  # Module setup, DO NOT EDIT, hook onto the 'setup' instead
         self.luciferManager = luciferManager
         self.actions = []
         self.isShellRun = ShellRun
         self.get_shell()
+        self.setup()
 
-    def run(self):
+    def setup(self):  # This is a setup hook which will be run on module load
+        pass
+
+    def run(self):  # This is a run hook and will be run on the module 'run'
         for action in self.actions:
             action()
 
-    def set_vars(self):
+    def set_vars(self):  # This is set_vars hook and will return the vars to set when the command 'set_vars' is run
         new_vars = {
         }
         return new_vars
 
-    def get_description(self):
+    def get_description(self):   # This is the description hook, returns a description on command 'description'
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/alexspam9090/Lucifer/pullRequest?prid=6402810)